src/Compilers/CSharp/Test/Syntax/Parsing/AttributeParsingTests.cs

```diff
@@ -0,0 +1,7033 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    [CompilerTrait(CompilerFeature.Attributes)]
+    public class AttributeParsingTests : ParsingTests
+    {
+        public AttributeParsingTests(ITestOutputHelper output) : base(output) { }
+
+        [Fact]
+        public void AttributeOnBlock()
```

Comment:
i enumerated all the cases i added support for, as well a lot of interesting corner cases.  For expression-statemnets, i also enumerated all expression forms.  

Nearly all cases we do a good job.  There are several cases we do a poor job on.  But we were poor there before, so it's not a regression.